2.0/contrib/etc/puma.cfg

```diff
@@ -1,4 +1,36 @@
+def get_max_memory()
+    if ENV.has_key?('MEMORY_LIMIT_IN_BYTES')
+        return ENV['MEMORY_LIMIT_IN_BYTES'].to_i
+    end
+
+    # Assume unlimited memory. 0.size is the number of bytes a ruby
+    # Fixnum class can hold. One bit is used for sign and one is used
+    # by Ruby to determine whether it's a number or pointer to an object.
+    # That's why we subtract two bits. This expresion should therefore be
+    # the largest signed Fixnum possible.
+    (2 ** (8*0.size - 2) - 1)
+end
+
+def get_min_threads()
+    ENV.fetch('PUMA_MIN_THREADS', '0').to_i
+end
+
+def get_max_threads()
+    ENV.fetch('PUMA_MAX_THREADS', '16').to_i
+end
+
+def get_workers()
+    base_memory = 50 * 1024 * 1024
+    per_worker_base_memory = 15 * 1024 * 1024
+    per_thread_memory = 103 * 1024 # roughly 0.1 MiB
+
+    cores = ENV.fetch('NUMBER_OF_CORES', '1').to_i
```

Comment:
I would need to reimplement it in every s2i language, that's why I did it
in awk

On Mon, Mar 14, 2016, 18:46 Michal Fojtik <notifications@github.com> wrote:

> In 2.0/contrib/etc/puma.cfg
> <https://github.com/openshift/sti-ruby/pull/96#discussion_r56043523>:
>
> > +end
> > +
> > +def get_min_threads()
> > +    ENV.fetch('PUMA_MIN_THREADS', '0').to_i
> > +end
> > +
> > +def get_max_threads()
> > +    ENV.fetch('PUMA_MAX_THREADS', '16').to_i
> > +end
> > +
> > +def get_workers()
> > +    base_memory = 50 * 1024 * 1024
> > +    per_worker_base_memory = 15 * 1024 * 1024
> > +    per_thread_memory = 103 * 1024 # roughly 0.1 MiB
> > +
> > +    cores = ENV.fetch('NUMBER_OF_CORES', '1').to_i
>
> i wonder if you can't do the math here (reading the file with ruby and
> parsing the value... would be much nicer than doing so with awk ;-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/sti-ruby/pull/96/files#r56043523>.
>